assets/js/components/billing/PurchaseCreditModal.jsx

```diff
@@ -69,235 +77,287 @@ class PurchaseCreditModal extends Component {
-      })
+      this.card.on("focus", () => {
+        this.setState({ paymentMethodSelected: undefined });
+      });
     }
 
-    this.props.getRouterAddress()
-    .then(({data}) => {
+    this.props.getRouterAddress().then(({ data }) => {
       if (typeof data.address === "string") {
-        this.setState({ routerAddress: data.address })
+        this.setState({ routerAddress: data.address });
       }
-    })
+    });
   }
 
   componentDidUpdate(prevProps) {
-    if(!prevProps.open && this.props.open) {
-      analyticsLogger.logEvent("ACTION_OPEN_PURCHASE_DC_MODAL")
+    if (!prevProps.open && this.props.open) {
+      analyticsLogger.logEvent("ACTION_OPEN_PURCHASE_DC_MODAL");
       if (this.state.countDC) {
-        this.setState({ gettingPrice: true, hntToBurn: null })
-        this.getOraclePrice()
+        this.setState({ gettingPrice: true, hntToBurn: null });
+        this.getOraclePrice();
       }
     }
 
-    if(prevProps.open && !this.props.open) {
-      this.card && this.card.unmount()
+    if (prevProps.open && !this.props.open) {
+      this.card && this.card.unmount();
     }
   }
 
   getOraclePrice = debounce(() => {
-    get('/api/data_credits/get_hnt_price')
-    .then(({data}) => {
-      const dcPrice = this.state.countDC * 0.00001
-      const hntPrice = data.price / 100000000
-      const hntToBurn = (Math.ceil((dcPrice / hntPrice) * 100000000) / 100000000).toFixed(8)
-
-      const nextTimeStamp = data.next_price_timestamp * 1000
-
-      this.setState({ gettingPrice: false, hntToBurn, nextTimeStamp })
-    })
-    .catch(() => {
-      // failed to get price, do not allow burn to continue
-      this.setState({ gettingPrice: false })
-    })
-  }, 500)
+    get("/api/data_credits/get_hnt_price")
+      .then(({ data }) => {
+        const dcPrice = this.state.countDC * 0.00001;
+        const hntPrice = data.price / 100000000;
+        const hntToBurn = (
+          Math.ceil((dcPrice / hntPrice) * 100000000) / 100000000
+        ).toFixed(8);
+
+        const nextTimeStamp = data.next_price_timestamp * 1000;
+
+        this.setState({ gettingPrice: false, hntToBurn, nextTimeStamp });
+      })
+      .catch(() => {
+        // failed to get price, do not allow burn to continue
+        this.setState({ gettingPrice: false });
+      });
+  }, 500);
 
   handleCountInputUpdate = (e) => {
-    if (e.target.value < 0) return
-    if (e.target.value.length > 11) return
+    if (e.target.value < 0) return;
+    if (e.target.value.length > 11) return;
     // Refactor out conversion rates between USD, DC, Bytes later
-    if (e.target.name == 'countDC') {
+    if (e.target.name == "countDC") {
       this.setState({
         countDC: e.target.value,
-        countUSD: e.target.value / 100000,
+        countUSD:
+          e.target.value * 0.00001 +
+          e.target.value *
+            0.00001 *
+            (window.host_dc_charge_percent ||
+              process.env.HOST_DC_CHARGE_PERCENT ||
+              0),
```

Comment:
main change

Reply:
we probably just need: 

window.host_dc_charge_percent || 0